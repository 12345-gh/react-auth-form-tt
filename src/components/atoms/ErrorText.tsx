import React from "react";
import styled from "styled-components";

const StyledErrorText = styled.div`
    font-family: Moderat;
    font-size: 16px;
    font-weight: 400;
    line-height: 22px;
    letter-spacing: -0.24888889491558075px; // ^ ,, ^
    text-align: left;
    color: #ed5f59;
    padding-top: 10px;
    padding-left: 27px;
`;

interface ErrorProps {
    message: string;
}

export const ErrorText: React.FC<ErrorProps> = ({ message }) => {
    return <StyledErrorText>{message}</StyledErrorText>;
};
