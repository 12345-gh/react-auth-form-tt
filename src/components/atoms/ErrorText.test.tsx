import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import { ErrorText } from "./ErrorText";

describe("ErrorText component", () => {
    it("matches the snapshot", () => {
        const { asFragment } = render(<ErrorText message="Snapshot Test" />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders the message correctly", () => {
        render(<ErrorText message="This is an error message" />);
        expect(screen.getByText("This is an error message")).toBeInTheDocument();
    });
});
