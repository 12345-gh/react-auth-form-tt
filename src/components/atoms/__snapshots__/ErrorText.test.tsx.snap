// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorText component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-braxZu kwHjtR"
  >
    Snapshot Test
  </div>
</DocumentFragment>
`;
